tree.go

```diff
@@ -1368,19 +1397,37 @@ func (n *LeafNode) GetProofItems(keys keylist, _ NodeResolverFn) (*ProofElements
 	if err := StemFromBytes(&poly[1], n.stem); err != nil {
 		return nil, nil, nil, fmt.Errorf("error serializing stem '%x': %w", n.stem, err)
 	}
-	if err := banderwagon.BatchMapToScalarField([]*Fr{&poly[2], &poly[3]}, []*Point{n.c1, n.c2}); err != nil {
-		return nil, nil, nil, fmt.Errorf("batch mapping to scalar fields: %s", err)
-	}
```

Comment:
OK, so here's a vital correctness fix.

We can't **always** do this here. If we're calling this `GetProofItems(...)` in a stateless reconstructed tree, then `n.c1` and `n.c2` will be `nil`. (Remember that now we only set the `isPOAStub` marker; so we can't blindly assume the data is there).

Reply:
right, in 402 I catch this even at the call site in `(*InternalNode)` so that I don't have to come here. What I don't see, is why you need to move this code instead of just using an `if` like you do anyway line 1421 ?

Reply:
you can still check the invariant later, but apart from that I don't quite see what that changes :thinking: ?

Reply:
> right, in 402 I catch this even at the call site in `(*InternalNode)` so that I don't have to come here.

Personally, I don't like that solution since if someone calls `(*LeafNode) GetProofItems(...)` that code would do the incorrect thing. You can say: "but usually, `GetProofItems(...)` gets call through an InternalNode". While that's true, there's nothing that prevents the caller to have some LeafNode and call this API. (As we do in many tests even). 

> What I don't see, is why you need to move this code instead of just using an `if` like you do anyway line 1421 ?

It's true that I didn't _need_ to do that, but I did so I can put in the same if the extra checks in the `else` case below. Also, putting this earlier wasn't needed since it's use later in the logic.

In any case, I can move it back and have two separate `ifs`.